packages/txservice/src/error.ts

```diff
@@ -221,11 +221,13 @@ export const parseError = (error: any): NxtpError => {
   };
 
   if (message.match(/execution reverted/)) {
-    throw new TransactionReverted(TransactionReverted.reasons.ExecutionFailed, undefined, context);
+    return new TransactionReverted(TransactionReverted.reasons.ExecutionFailed, undefined, context);
   } else if (message.match(/always failing transaction/)) {
-    throw new TransactionReverted(TransactionReverted.reasons.AlwaysFailingTransaction, undefined, context);
+    return new TransactionReverted(TransactionReverted.reasons.AlwaysFailingTransaction, undefined, context);
   } else if (message.match(/gas required exceeds allowance/)) {
-    throw new TransactionReverted(TransactionReverted.reasons.GasExceedsAllowance, undefined, context);
+    return new TransactionReverted(TransactionReverted.reasons.GasExceedsAllowance, undefined, context);
+  } else if (message.match(/tx doesn't have the correct nonce|another transaction with same nonce|same hash was already imported|transaction nonce is too low|nonce too low/)) {
+    return new AlreadyMined(AlreadyMined.reasons.NonceExpired, context);
```

Comment:
This should match all nonce messages better. Learned today that they can come back as a SERVER_ERROR. In particular, they come back like this from ganache run locally

(Also the above `throw`s were a bug, changed to `return`